### PR TITLE
[feat]: Add Exchange Page

### PR DESCRIPTION
**Pull request by dhruv1294**
Added exchange functionality using global streams

TODO: Better Exchange page UI and bottom sheet UI

### CODE REVIEW

File: lib/pages/exchange_page.dart [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+        return const Center(
+          child: CircularProgressIndicator(
+            color: secondaryColor,
+          ),
+        );
+      }
+    });
+  }
+
+  Container _stockExchangeItem(
```
Comment (Sudhindra3000) [inline lib/pages/exchange_page.dart:201]: Make this as a separate widget

**Review by Sudhindra3000 — APPROVED** [review]

File: lib/blocs/exchange/exchange_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+part 'exchange_state.dart';
+
+class ExchangeCubit extends Cubit<ExchangeState> {
+  ExchangeCubit() : super(const ExchangeLoading());
+
+  Future<void> listenToExchangeStream(Map<int, Stock> mapOfStocks) async {
+    // Load initial state
+    emit(ExchangeDataLoaded(mapOfStocks
+        .map((key, value) => MapEntry(key, stockDataToExchangeData(value)))));
+    final exchangeStream = getIt<GlobalStreams>().stockExchangeStream;
```
Comment (satyamurti) [inline lib/blocs/exchange/exchange_cubit.dart:19]: So this stream will start at the start of app?

**Review by satyamurti — CHANGES_REQUESTED** [review]

File: lib/blocs/exchange/exchange_cubit.dart (hunk 2 above)
Comment (dhruv1294) [reply lib/blocs/exchange/exchange_cubit.dart:19]: yes ig since its been made a global stream

Comment (dhruv1294) [reply lib/blocs/exchange/exchange_cubit.dart:19]: coz the stocklist map will need to be updated from this stream as well

**Review by satyamurti — APPROVED** [review]

**Pull request merged**